structures/auction.js

```diff
@@ -44,12 +44,12 @@ exports.auction = function (config = null) {
       getitnow: s.optional(s.number()),
       winningbid: s.optional(Uuid),
       puPlace: s.dynamic((v, p) => { return v.length === 5 ? Place : PlaceTZ }),
-      puLocation: s.optional(s.union([s.size(s.array(s.size(s.string(), 2, 32)), 0, 2), s.size(s.string(), 2, 64)])),
+      puLocation: s.optional(s.union([s.size(s.array(s.pattern(s.string(), /(-)?\d+\.\d+/)), 0, 2), s.pattern(s.string(), /(-)?\d+\.\d+,(-)?\d+\.\d+/)])),
```

Comment:
Can be abstracted I think.

Reply:
@fanshan : can you please explain a bit how you would abstract it ?